src/electron/platform/android/setup/android-setup-start-listener.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { FeatureFlagStore } from 'background/stores/global/feature-flag-store';
+import { UserConfigurationStore } from 'background/stores/global/user-configuration-store';
+import { UnifiedFeatureFlags } from 'electron/common/unified-feature-flags';
+import { AndroidSetupActionCreator } from 'electron/flux/action-creator/android-setup-action-creator';
+import { AndroidSetupStore } from 'electron/flux/store/android-setup-store';
+
+export class AndroidSetupStartListener {
+    constructor(
+        private readonly userConfigStore: UserConfigurationStore,
+        private readonly androidSetupStore: AndroidSetupStore,
+        private readonly featureFlagStore: FeatureFlagStore,
+        private androidSetupActionCreator: AndroidSetupActionCreator,
+    ) {}
+
+    public initialize(): void {
+        this.storeChanged();
+
+        this.userConfigStore.addChangedListener(this.storeChanged);
+        this.featureFlagStore.addChangedListener(this.storeChanged);
+        this.androidSetupStore.addChangedListener(this.storeChanged);
+    }
+
+    private storeChanged = () => {
+        const stepId = this.androidSetupStore.getState().currentStepId;
+        const featureEnabled = this.featureFlagStore.getState()[UnifiedFeatureFlags.adbSetupView];
```

Comment:
if someone is in the middle of a different workflow, turning the feature flag on will begin ADB setup regardless of where the UI is